### PR TITLE
cli: allow branch list to combine -r and -a

### PR DESCRIPTION
**Pull request by Kintaro**
As discussed on discord  [here](https://discord.com/channels/968932220549103686/969291218347524238/1221084605617279016), the change allows `--all` to be used with `-r` when listing branches.

Also discussed was renaming `--all` to `--all-remotes`, but this change is a bit longer and will be followed in a separate PR.

# Checklist

If applicable:
- [X] I have updated `CHANGELOG.md`
- [ ] I have updated the documentation (README.md, docs/, demos/)
- [ ] I have updated the config schema (cli/src/config-schema.json)
- [X] I have added tests to cover my changes

### CODE REVIEW

File: cli/tests/test_branch_command.rs [hunk 1]
```diff
@@ -1388,6 +1388,35 @@ fn test_branch_list_conflicted() {
     "###);
 }
 
+#[test]
+fn test_branch_list_revset_all() {
+    let test_env = TestEnvironment::default();
+    test_env.add_config("git.auto-local-branch = true");
+
+    // Initialize remote refs
```
Comment (yuja) [inline cli/tests/test_branch_command.rs:1396]: nit: copy-paste error? it doesn't initialize remote refs.

File: cli/src/commands/branch.rs [hunk 2]
```diff
@@ -98,7 +98,7 @@ pub struct BranchDeleteArgs {
 pub struct BranchListArgs {
     /// Show all tracking and non-tracking remote branches including the ones
     /// whose targets are synchronized with the local branches.
-    #[arg(long, short, conflicts_with_all = ["names", "revisions", "tracked"])]
+    #[arg(long, short, conflicts_with_all = ["names", "tracked"])]
```
Comment (yuja) [inline cli/src/commands/branch.rs:101]: > Also discussed was renaming `--all` to `--all-remotes`, but this change is a bit longer and will be followed in a separate PR.

Perhaps, `--all` can be kept as a hidden alias (if you mean it's not simple to update all tests and add deprecation handling.)

**Review by yuja — APPROVED** [review]

File: cli/tests/test_branch_command.rs (hunk 1 above)
Comment (yuja) [reply cli/tests/test_branch_command.rs:1396]: and the test can be moved to `test_branch_list_filtered()`?

File: cli/src/commands/branch.rs (hunk 2 above)
Comment (Kintaro) [reply cli/src/commands/branch.rs:101]: It's not too difficult, just a bit noisy PR. I can add an alias, sure

File: cli/tests/test_branch_command.rs (hunk 1 above)
Comment (Kintaro) [reply cli/tests/test_branch_command.rs:1396]: Ah, if it's ok as part of filtered, I'll move in there.
